entities/entities/spawned_shipment/init.lua

```diff
@@ -108,6 +108,25 @@ function ENT:Use(activator, caller)
     end)
 end
 
+local function calculateAmmo(class, shipment)
+    local clip1, ammoadd = shipment.clip1, shipment.ammoadd
+
+    local defaultClip, clipSize
+    local wep_tbl = weapons.Get(class)
+    if wep_tbl and wep_tbl.Primary then
```

Comment:
When this condition is `false`, `defaultClip` and `clipSize` will remain `nil`. That will cause errors with the `math.min` call and arithmetic below.

Reply:
Weapon.Primary does not have to be a table. Type check before indexing.